types/errors_test.go

```diff
@@ -50,8 +48,12 @@ func (s testErrorSuite) TestError(c *C) {
 		ErrInvalidWeekModeFormat,
 		ErrWrongValue,
 	}
+
+	t.Parallel()
 	for _, err := range kvErrs {
-		code := terror.ToSQLError(err).Code
-		c.Assert(code != mysql.ErrUnknown && code == uint16(err.Code()), IsTrue, Commentf("err: %v", err))
+		t.Run("ValidateErrorCodes", func(t *testing.T) {
```

Comment:
You don't need `t.Run("ValidateErrorCodes", func(t *testing.T) {` if there is `t.Parallel()` outside. You could refer to the document of go for more description, how sub tests and parallel tests work.

Reply:
We create sub tests to parallel tests.